eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/BackPropagatedRSWriter.java

```diff
@@ -241,33 +168,7 @@ else if (ssInRemappedAssemblyGroupedByID.containsKey(ssIDToBeClustered)) {
         }
     }
 
-    private void handleClusteredVariantInserts(BulkOperations evaCVEInserts, BulkOperations dbsnpCVEInserts,
-                                               int numEVARSInserts, int numDbsnpRSInserts) {
-        try {
-            if (numEVARSInserts > 0) {
-                evaCVEInserts.execute();
-            }
-
-            if (numDbsnpRSInserts > 0) {
-                dbsnpCVEInserts.execute();
-            }
-        }
-        catch (DuplicateKeyException duplicateKeyException) {
-            MongoBulkWriteException writeException = ((MongoBulkWriteException) duplicateKeyException.getCause());
-            // Consider a scenario where we are back-propagating rs ID:
-            // RS1 corresponding to SS1 from ASM2 (remapped assembly) to ASM1 (original assembly)
-            // If the hash for back-propagated RS1 already exists in ASM1,
-            // it means that, for some reason, SS1 was not clustered with RS1 when ASM1 underwent clustering.
-            // It is better to flag such issues for later analysis.
-            extractUniqueHashesForDuplicateKeyError(writeException)
-                    .forEach(hash -> logger.error("Attempted to insert RS record with already existing hash " + hash +
-                                                         "during RS ID back-propagation! This could be symptomatic " +
-                                                         "of clustering issues with the original assembly "
-                                                         + this.originalAssembly + "."));
-        }
-    }
-
-    private ClusteredVariantEntity getSuitableRSFromRemappedSS(
+    private Long getSuitableRSFromRemappedSS(
```

Comment:
I'm not sure I see the point of this function. There should only be one `submittedVariantEntity per ssid in the remapped assembly

Reply:
Until we resolve the existing issues (see EVA-2830 and EVA-2840) that could lead to multiple SS, we might have to keep this around to have a predictable backpropagation.